cs/src/comm/epoxy-transport/EpoxyTransport.cs

```diff
@@ -199,6 +206,13 @@ public struct TimeoutConfig
             logger = new Logger(logSink, enableDebugLogs);
             // Metrics sink may be null
             metrics = new Metrics(metricsSink);
+
+            shutdownLock = new ReaderWriterLockSlim(LockRecursionPolicy.NoRecursion);
+            isShutdown = false;
+
+            collectionsLock = new object();
```

Comment:
After switching everything over to a `CleanupCollection`, there is no longer even shutdown state in the `EpoxyTransport` that need guarding.